File: Scripts/results/generated_code_results/Flutter/bak_code/Wellness/Headspace/CalmMethodsScreen.dart
import 'package:flutter/material.dart';
class CalmMethodsScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pushNamed(context, 'Explore'),
),
title: Text('Calming Everyday Anxiety'),
actions: [
Icon(Icons.wifi),
SizedBox(width: 10),
Icon(Icons.battery_full),
],
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: EdgeInsets.all(16.0),
child: Text(
'Get in-the-moment support for anxious thinking, plus expert guidance for cultivating a calmer mind every day.',
style: TextStyle(fontSize: 16),
),
),
Padding(
padding: EdgeInsets.symmetric(horizontal: 16.0),
child: Text(
'SOS for Anxious Moments',
style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
),
),
GridView.count(
shrinkWrap: true,
physics: NeverScrollableScrollPhysics(),
crossAxisCount: 2,
padding: EdgeInsets.all(16),
children: [
_buildFeatureCard(
'Feeling Overwhelmed SOS',
'Meditation · 3-4 min',
'Give yourself room to breathe.',
'assets/images/template.png',
),
_buildFeatureCard(
'Panicking SOS',
'Meditation · 3 min',
'Anchor your mind and body in the present.',
'assets/images/template.png',
),
_buildFeatureCard(
'Breathe',
'Meditation · 1-3 min',
'Add a sense of spaciousness to your day.',
'assets/images/template.png',
),
_buildFeatureCard(
'Reset',
'Meditation · 3-10 min',
'Find some focus and relaxation during a busy day.',
'assets/images/template.png',
),
],
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Today'),
BottomNavigationBarItem(icon: Icon(Icons.explore), label: 'Explore'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: '子骆'),
],
onTap: (index) {
if (index == 1) {
Navigator.pushNamed(context, 'Explore');
}
},
),
);
}
Widget _buildFeatureCard(String title, String subtitle, String description, String imagePath) {
return Card(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Expanded(
child: Image.asset(
imagePath,
fit: BoxFit.cover,
width: double.infinity,
),
),
Padding(
padding: EdgeInsets.all(8.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
title,
style: TextStyle(fontWeight: FontWeight.bold),
),
SizedBox(height: 4),
Text(subtitle),
SizedBox(height: 4),
Text(
description,
style: TextStyle(fontSize: 12),
),
],
),
),
],
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Wellness/Headspace/CalmMethodsScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity, ImageBackground } from 'react-native';

const CalmMethodsScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.timerText}>00:43</Text>
        <Text style={styles.batteryText}>94%</Text>
        <Image source={require('../assets/snack-icon.png')} style={styles.signalIcon} />
        <Image source={require('../assets/snack-icon.png')} style={styles.wifiIcon} />
        <Image source={require('../assets/snack-icon.png')} style={styles.batteryIcon} />
      </View>
      <View style={styles.content}>
        <Text style={styles.title}>Calming Everyday Anxiety</Text>
        <Text style={styles.description}>
          Get in-the-moment support for anxious thinking, plus expert guidance for cultivating a calmer mind every day.
        </Text>
        <Text style={styles.sectionTitle}>SOS for Anxious Moments</Text>
        <View style={styles.sosContainer}>
          <TouchableOpacity style={styles.sosItem}>
            <ImageBackground source={require('../assets/snack-icon.png')} style={styles.sosImage}>
              <Text style={styles.sosTitle}>Feeling Overwhelmed SOS</Text>
              <Text style={styles.sosDetails}>Meditation · 3-4 min</Text>
              <Text style={styles.sosDescription}>Give yourself room to breathe.</Text>
            </ImageBackground>
          </TouchableOpacity>
          <TouchableOpacity style={styles.sosItem}>
            <ImageBackground source={require('../assets/snack-icon.png')} style={styles.sosImage}>
              <Text style={styles.sosTitle}> panicking SOS</Text>
              <Text style={styles.sosDetails}>Meditation · 3 min</Text>
              <Text style={styles.sosDescription}>Anchor your mind and body in the present.</Text>
            </ImageBackground>
          </TouchableOpacity>
        </View>
        <View style={styles.sosContainer}>
          <TouchableOpacity style={styles.sosItem}>
            <ImageBackground source={require('../assets/snack-icon.png')} style={styles.sosImage}>
              <Text style={styles.sosTitle}>Breathe</Text>
              <Text style={styles.sosDetails}>Meditation · 1-3 min</Text>
              <Text style={styles.sosDescription}>Add a sense of spaciousness to your day.</Text>
            </ImageBackground>
          </TouchableOpacity>
          <TouchableOpacity style={styles.sosItem}>
            <ImageBackground source={require('../assets/snack-icon.png')} style={styles.sosImage}>
              <Text style={styles.sosTitle}>Reset</Text>
              <Text style={styles.sosDetails}>Meditation · 3-10 min</Text>
              <Text style={styles.sosDescription}>Find some focus and relaxation during a busy day.</Text>
            </ImageBackground>
          </TouchableOpacity>
        </View>
        <Text style={styles.unlockText}>Unlock the Headspace Library</Text>
      </View>
      <View style={styles.bottomNav}>
        <TouchableOpacity onPress={() => navigation.navigate('Home')}>
          <Text style={styles.navItem}>Today</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Explore')}>
          <Text style={styles.navItemActive}>Explore</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Profile')}>
          <Text style={styles.navItem}>子骆</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  header: {
    height: 60,
    backgroundColor: '#0000ff',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  timerText: {
    color: '#fff',
  },
  batteryText: {
    color: '#fff',
  },
  signalIcon: {
    width: 20,
    height: 20,
  },
  wifiIcon: {
    width: 20,
    height: 20,
  },
  batteryIcon: {
    width: 20,
    height: 20,
  },
  content: {
    flex: 1,
    padding: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#000',
    textAlign: 'center',
  },
  description: {
    fontSize: 16,
    color: '#000',
    textAlign: 'center',
    marginVertical: 10,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#000',
    marginVertical: 10,
  },
  sosContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: 10,
  },
  sosItem: {
    width: '48%',
    backgroundColor: '#f0f0f0',
    borderRadius: 10,
    overflow: 'hidden',
  },
  sosImage: {
    width: '100%',
    height: 120,
    justifyContent: 'flex-end',
    padding: 10,
  },
  sosTitle: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#fff',
  },
  sosDetails: {
    fontSize: 12,
    color: '#fff',
  },
  sosDescription: {
    fontSize: 12,
    color: '#fff',
  },
  unlockText: {
    fontSize: 16,
    color: '#000',
    textAlign: 'center',
    marginVertical: 20,
  },
  bottomNav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: 60,
    backgroundColor: '#000',
    position: 'absolute',
    bottom: 0,
    width: '100%',
  },
  navItem: {
    color: '#fff',
  },
  navItemActive: {
    color: '#ff0',
  },
});

export default CalmMethodsScreen;